spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/TestSparkBQLineageExtensionProvider.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2024 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.spark.bigquery;
+
+import io.openlineage.spark.extension.OpenLineageExtensionProvider;
+
+public class TestSparkBQLineageExtensionProvider implements OpenLineageExtensionProvider {
+  @Override
+  public String shadedPackage() {
+    return "com.google.cloud.spark.bigquery.repackaged.io.openlineage.spark.shade";
+  }
+
+  @Override
+  public String getVisitorClassName() {
+    return "io.openlineage.spark.shade.extension.v1.lifecycle.plan.SparkOpenLineageExtensionVisitor";
```

Comment:
Wouldn't this class be shaded? shouldn't it be `com.google.cloud.spark.bigquery.repackaged.io.openlineage.spark.shade.extension.v1.lifecycle.plan.SparkOpenLineageExtensionVisitor`?

Reply:
The issue is that, for tests, classes provided to the classpath are not shaded. Therefore, we need an alternative `OpenLineageExtensionProvider` that will point to the unshaded resource. Hence, we have introduced the `spark.openlineage.testExtensionProvider` config property to ensure that only one `OpenLineageExtensionProvider` is selected.

Reply:
Your can return `SparkOpenLineageExtensionVisitor.class.getCanonicalName()`, the import will change as part of the shading.

Reply:
you are right!